src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -1556,75 +1556,75 @@ public void MainWindowHandle_GetNotStarted_ThrowsInvalidOperationException()
             var process = new Process();
             Assert.Throws<InvalidOperationException>(() => process.MainWindowHandle);
         }
-        
-        [Fact]
-        [OuterLoop]
-        [Trait(XunitConstants.Category, XunitConstants.IgnoreForCI)] // Pops UI
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]  // no Notepad on Nano
+        [OuterLoop("Launches notepad")]
         [PlatformSpecific(TestPlatforms.Windows)]
-        public void MainWindowHandle_GetWithGui_ShouldRefresh_Windows()
```

Comment:
Any particular reason you removed the old tests -- were they flaky? They did at least verify the public properties.

Reply:
I've renamed the test and refactored it (moved the common logic to a separate method).

But if it's not obvious when you look at the code then it's likely that I've tried too hard with the refactor and it does not add any value?

Reply:
I wonder if you are trying to hard - it is a bit hard to read. Also, in the old test, we would always verify that MainWindowTitle turned to something other than empty (it should really test it's not null either) whereas now, depending on timing of process.MainWindowHandle turning non null, it may not read the public property at all because it will never enter the while loop?

If the old tests were stable, would it make sense to keep them but also extend them to read the field as well?

Reply:
I've reverted my refactoring